Src/coffeescript/cql-execution/src/datatypes/datetime.coffee

```diff
@@ -201,7 +204,17 @@ module.exports.DateTime = class DateTime
       a = new DateTime(a.year, a.month, a.day, a.hour, a.minute, a.second, 0, a.timezoneOffset)
       b = new DateTime(b.year, b.month, b.day, b.hour, b.minute, b.second, 0, b.timezoneOffset)
 
-    a.durationBetween(b, unitField)
+    if unitField == DateTime.Unit.YEAR || unitField == DateTime.Unit.MONTH
+      a.durationBetween(b, unitField)
+    else
+      a = a.toUncertainty()
```

Comment:
I think the new "a" and "b" variables should be named something different. Thoughts?